realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -81,7 +82,7 @@
  */
 
 @RealmClass
-public abstract class RealmObject {
+public abstract class RealmObject<E extends RealmObject> {
```

Comment:
Yes, the pattern is not unknown from RxJava itself: `Observable.from()`, `Observable.just()`, but it really really break up the flow when you are coding. I tried it when working on the examples.

Upside is that you don't actually *need* the generic, but if you don't add it you just get Object as the return type in the Observable chain.

I have been contemplating if we should add an explicit ListenerInterface, but that just seems like unnecessary boilerplate